components/script/dom/bindings/codegen/CodegenRust.py

```diff
@@ -3014,7 +3046,7 @@ def define(self):
         return CGAbstractMethod.define(self)
 
     def definition_body(self):
-        if self.descriptor.interface.isCallback():
+        if self.descriptor.interface.isCallback() or self.descriptor.interface.isNamespace():
```

Comment:
This change confuses me, since gecko's codegen has lots of asserts in constructor-related code that the descriptor interface is not a namespace.